packages/pigeon/run_tests.sh

```diff
@@ -86,195 +90,337 @@ test_pigeon_android() {
-###############################################################################
-pushd $PWD
-pub run pigeon \
+test_running_without_arguments() {
+  pub run pigeon 1>/dev/null
+}
+
+run_flutter_unittests() {
+  pushd $PWD
+  pub run pigeon \
     --input pigeons/message.dart \
     --dart_out platform_tests/flutter_null_safe_unit_tests/lib/null_safe_pigeon.dart
-cd platform_tests/flutter_null_safe_unit_tests
-flutter pub get
-flutter test test/null_safe_test.dart 
-popd
+  cd platform_tests/flutter_null_safe_unit_tests
+  flutter pub get
+  flutter test test/null_safe_test.dart
+  popd
+}
 
-###############################################################################
-# Mock handler flutter tests.
-###############################################################################
-pushd $PWD
-pub run pigeon \
-  --input pigeons/message.dart \
-  --dart_out mock_handler_tester/test/message.dart \
-  --dart_test_out mock_handler_tester/test/test.dart
-dartfmt -w mock_handler_tester/test/message.dart
-dartfmt -w mock_handler_tester/test/test.dart
-cd mock_handler_tester
-flutter test
-popd
+run_mock_handler_tests() {
+  pushd $PWD
+  pub run pigeon \
+    --input pigeons/message.dart \
+    --dart_out mock_handler_tester/test/message.dart \
+    --dart_test_out mock_handler_tester/test/test.dart
+  dartfmt -w mock_handler_tester/test/message.dart
+  dartfmt -w mock_handler_tester/test/test.dart
+  cd mock_handler_tester
+  flutter test
+  popd
+}
 
-###############################################################################
-# Compilation tests (Code is generated and compiled)
-###############################################################################
-# Make sure the artifacts are present.
-flutter precache
-# Make sure flutter dependencies are available.
-pushd $PWD
-cd e2e_tests/test_objc/
-flutter pub get
-popd
-test_pigeon_ios ./pigeons/async_handlers.dart
-test_null_safe_dart ./pigeons/message.dart
-test_pigeon_android ./pigeons/voidflutter.dart
-test_pigeon_android ./pigeons/voidhost.dart
-test_pigeon_android ./pigeons/host2flutter.dart
-test_pigeon_android ./pigeons/message.dart
-test_pigeon_android ./pigeons/void_arg_host.dart
-test_pigeon_android ./pigeons/void_arg_flutter.dart
-test_pigeon_android ./pigeons/list.dart
-test_pigeon_android ./pigeons/all_datatypes.dart
-test_pigeon_android ./pigeons/async_handlers.dart
-test_pigeon_ios ./pigeons/message.dart
-test_pigeon_ios ./pigeons/host2flutter.dart
-test_pigeon_ios ./pigeons/voidhost.dart
-test_pigeon_ios ./pigeons/voidflutter.dart
-test_pigeon_ios ./pigeons/void_arg_host.dart
-test_pigeon_ios ./pigeons/void_arg_flutter.dart
-test_pigeon_ios ./pigeons/list.dart
-test_pigeon_ios ./pigeons/all_datatypes.dart
+run_dart_compilation_tests() {
+  # Make sure the artifacts are present.
+  flutter precache
+  # Make sure flutter dependencies are available.
+  pushd $PWD
+  cd e2e_tests/test_objc/
+  flutter pub get
+  popd
+  test_pigeon_dart ./pigeons/all_datatypes.dart
```

Comment:
do we also need to add a warning somewhere saying if you add more tests, you must list them here?

Reply:
I don't think so.  When doing TDD you'll add the tests here as you implement them for the different generators.